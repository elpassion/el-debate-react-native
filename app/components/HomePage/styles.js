import { StyleSheet, Platform } from 'react-native'

export default styles = StyleSheet.create({
  description: {
    marginBottom: 20,
    fontSize: 20,
    textAlign: 'center',
    fontFamily: 'helvetica-neue',
    color: '#656565'
  },
  navContainer: {
    flex: 1
  },
  container: {
    alignItems: 'center',
    flex: 1,
    top: (Platform.OS === 'ios') ? 100 : 70,
  },
  backgroundImage: {
    position: 'absolute',
    bottom: (Platform.OS === 'ios') ? 50 : 0,
    width: '85%',
  },
  iosPinInput: {
    bottom: 0,
    height: 40,
    width: '80%',
    textAlign: 'center',
    marginRight: 5,
    marginTop: 40,
    borderBottomColor: '#C0C0C0',
    borderBottomWidth: 1,
    fontSize: 18,
    color: '#8F8F8F',
  },
  androidPinInput: {
    bottom: 0,
    height: 60,
    width: '80%',
    textAlign: 'center',
    marginRight: 5,
    marginTop: 40,
    fontSize: 20,
    color: '#8F8F8F',
  },
  pinButton: {
    alignSelf: 'center',
    height: (Platform.OS === 'ios') ? 40 : 50,
    width: '80%',
    bottom: (Platform.OS === 'ios') ? 120 : 80,
    backgroundColor: '#4CC359',
    borderRadius: 0,
    borderColor: '#fff',
  },
  footer: {
    position: 'absolute',
    bottom: (Platform.OS === 'ios') ? 0 : 50,
    right: 0,
    left: 0,
    alignItems: 'center'
  }
})
