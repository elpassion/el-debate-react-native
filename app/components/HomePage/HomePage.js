import React, { Component } from 'react';
import {
  Text,
  TextInput,
  View,
  ActivityIndicator,
  Image,
  Alert,
  Platform,
  AsyncStorage
} from 'react-native';
import Button from 'apsl-react-native-button'
import styles from './styles'
import { Redirect, Route } from "react-router-native";
import NavigationBar from 'react-native-navigation-bar';
import Api from '/app/api/Api'

export default class HomePage extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isLoading: false,
      message: '',
      searchString: '',
      isFetched: false
    };
  }

  _handleResponse = (response) => {
    if (response.error) {
      this.setState({ message: response.error });
    } else {
      this.setState({ isFetched: true })
      AsyncStorage.setItem('authToken', response.auth_token)
    }
  };

  _onLoginPressed = () => {
    this.setState({ isLoading: true, message: '' });
    AsyncStorage.setItem('code', this.state.searchString)
    Api.login(this.state.searchString)
      .then(response => response.json())
      .then(response => this._handleResponse(response))
      .catch(error =>
        Alert.alert('Something went wrong' + error)
    );
    this.setState({ isLoading: false });
  };

  _onSearchTextChanged = (event) => {
    this.setState({ searchString: event.nativeEvent.text });
  };

  componentDidMount = () => {
     AsyncStorage.getItem('code').then((value) => this.setState({ 'searchString': value }))
  }

  render() {
    const DetailsRoute = () => (
      this.state.isFetched ? <Redirect push to="/debate-details"/> : null
    )
    const CodeInput =
      this.state.isLoading ?
      <View style={{marginTop: 40}}>
        <ActivityIndicator size='large'/>
      </View> :
      <TextInput
        keyboardType = {(Platform.OS === 'ios') ? 'default' : 'numeric'}
        maxLength = {5}
        style={(Platform.OS === 'ios') ? styles.iosPinInput : styles.androidPinInput}
        value={this.state.searchString}
        onChange={this._onSearchTextChanged}
        placeholder='EL Debate PIN'
        returnKeyType = 'done'
        onSubmitEditing={this._onLoginPressed}/>;
    return (
      <View style={styles.navContainer}>
        <NavigationBar
          title={'EL Debate'}
          height={50}
          titleColor={'#fff'}
          backgroundColor={'#4CC359'}
        />
      <View style={styles.container}>
        <Text style={styles.description}>{this.state.message}</Text>
        <Text style={styles.description}>
          Welcome to
        </Text>
        <Image source={require('/resources/images/logo.png')}/>
        {CodeInput}
        <View style={styles.footer}>
          <Image source={require('/resources/images/backgroundimg.png')} style={styles.backgroundImage} />
          <Button
            onPress={this._onLoginPressed}
            style={styles.pinButton}
            textStyle={{ color: '#fff', fontFamily: 'helvetica-neue', fontSize: 14 }}
          >
          Log in
          </Button>
          <DetailsRoute/>
        </View>
        </View>
      </View>
    );
  }
}
