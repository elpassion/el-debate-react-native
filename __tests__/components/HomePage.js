import 'react-native';
import React from 'react';

import HomePage from 'app/components/HomePage/HomePage';

import { shallow, mount, render } from 'enzyme';

import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });


it('renders correctly', () => {
  const tree = shallow(<HomePage />)
  expect(tree).toMatchSnapshot();
});
