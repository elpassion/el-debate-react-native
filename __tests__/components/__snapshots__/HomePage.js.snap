// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <NavigationBar
    backgroundColor="#4CC359"
    height={50}
    leftButtonTitle={null}
    leftButtonTitleColor="#000"
    rightButtonTitle={null}
    rightButtonTitleColor="#000"
    title="EL Debate"
    titleColor="#fff"
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "top": 100,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#656565",
          "fontFamily": "helvetica-neue",
          "fontSize": 20,
          "marginBottom": 20,
          "textAlign": "center",
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#656565",
          "fontFamily": "helvetica-neue",
          "fontSize": 20,
          "marginBottom": 20,
          "textAlign": "center",
        }
      }
    >
      Welcome to
    </Text>
    <Image
      source={1}
    />
    <TextInput
      keyboardType="default"
      maxLength={5}
      onChange={[Function]}
      onSubmitEditing={[Function]}
      placeholder="EL Debate PIN"
      returnKeyType="done"
      style={
        Object {
          "borderBottomColor": "#C0C0C0",
          "borderBottomWidth": 1,
          "bottom": 0,
          "color": "#8F8F8F",
          "fontSize": 18,
          "height": 40,
          "marginRight": 5,
          "marginTop": 40,
          "textAlign": "center",
          "width": "80%",
        }
      }
      value=""
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "bottom": 50,
            "position": "absolute",
            "width": "85%",
          }
        }
      />
      <Button
        onPress={[Function]}
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "#4CC359",
            "borderColor": "#fff",
            "borderRadius": 0,
            "bottom": 120,
            "height": 40,
            "width": "80%",
          }
        }
        textStyle={
          Object {
            "color": "#fff",
            "fontFamily": "helvetica-neue",
            "fontSize": 14,
          }
        }
      >
        Log in
      </Button>
      <DetailsRoute />
    </View>
  </View>
</View>
`;
